cdc/sink/dmlsink/txn/mysql/mysql.go

```diff
@@ -138,13 +137,11 @@ func NewMySQLBackends(
 		// Because each connection can not hold at lease one prepared statement.
 		if maxPreparedStmtCount == 0 || maxPreparedStmtCount/(cfg.WorkerCount+1) == 0 {
 			cachePrepStmts = false
-		} else if maxPreparedStmtCount/(cfg.WorkerCount+1) < prepStmtCacheSize {
-			// if maxPreparedStmtCount/(cfg.WorkerCount+1) < prepStmtCacheSize,
-			// it means that the prepared statement cache is too large on clientsize.
-			// adjust the size of the prepared statement cache on clientsize.
-			// to avoid error `Can't create more than max_prepared_stmt_count statements`
-			prepStmtCacheSize = maxPreparedStmtCount / (cfg.WorkerCount + 1)
```

Comment:
https://github.com/pingcap/tiflow/pull/8387 has tried to fixed related issue, seems the `prepStmtCacheSize` is not set to a proper value.

Reply:
The solution can't resolve the situation with more than one CDC instances. So I remove `cacheSize`. Now cache evict is triggered by errors returned by `db.Prepare`.

Reply:
although this does not fix the situation with more than one CDC instances. It set the reasonable limit for the cache size. This value is much lower than `16*1024` when there is more than one worker count. If we hard code the cache size with `16*1024`, we might need to keep try db.Prepare() then got error and fallback to query.